### PR TITLE
Added masturbating bool, consensual bool and lock bool

### PR DESCRIPTION
**Pull request by kayfour**

### CODE REVIEW

File: AAUnlimited/External/ExternalClasses/CharacterActivity.h [hunk 1]
```diff
@@ -28,8 +29,10 @@ class CharacterActivity {
 	DWORD m_lastConversationSuccess; //(3=trumpet, 2 = nice, 1 = normal)
 	BYTE m_unknown7[4];
 	DWORD m_lastConversationAnswerPercent;
-	DWORD m_lastConversationAnswer;
-	BYTE m_unknown8[0x3C];
+	DWORD m_lastConversationAnswer; //52
```
Comment (geneishouko) [inline AAUnlimited/External/ExternalClasses/CharacterActivity.h]: Consistency: Change these to hexadecimal (0x###) notation

File: AAUnlimited/External/ExternalClasses/CharacterActivity.h [hunk 2]
```diff
@@ -28,8 +29,10 @@ class CharacterActivity {
...
+	BYTE m_unknown11[12];
+	DWORD m_isMasturbating;
+	BYTE m_unknown8[44];
```
Comment (geneishouko) [inline AAUnlimited/External/ExternalClasses/CharacterActivity.h]: Hexadecimal notation

File: AAUnlimited/Functions/Shared/Triggers/Thread.h [hunk 3]
```diff
@@ -155,6 +158,8 @@ namespace Shared {
 			Value GetCardIntelligenceValue(std::vector<Value>& params); //int(int)
 			Value GetCardIntelligenceRank(std::vector<Value>& params); //int(int)
 			Value GetCardStrength(std::vector<Value>& params); //int(int)
+			Value GetCardLocked(std::vector<Value>& params);
+			Value GetCardFapping(std::vector<Value>& params);
```
Comment (geneishouko) [inline AAUnlimited/Functions/Shared/Triggers/Thread.h]: (!) SetMasturbating 
Get and Set use different terms

File: AAUnlimited/Functions/Shared/Triggers/Expressions.cpp [hunk 4]
```diff
@@ -458,6 +458,25 @@ namespace Shared {
...
+		Value Thread::GetCardLocked(std::vector<Value>& params) {
+			int card = params[0].iVal;
+			CharInstData* cardInst = &AAPlay::g_characters[card];
+			if (!cardInst->IsValid()) return Value(TEXT(""));
+
+			return Value(cardInst->m_char->m_moreData1->m_activity->m_interactionLock);
+		}
+		//int(int)
+
+		Value Thread::GetCardFapping(std::vector<Value>& params) {
```
Comment (geneishouko) [inline AAUnlimited/Functions/Shared/Triggers/Expressions.cpp]: (!) SetMasturbating
Get and Set use different terms

File: AAUnlimited/Functions/Shared/Triggers/Expressions.cpp [hunk 5]
```diff
@@ -2224,6 +2243,19 @@ namespace Shared {
...
 				},
+				{
+					93, EXPRCAT_CHARPROP,
+					TEXT("Locked state"), TEXT("%p ::LockState"), TEXT("Returns whether the character has the red circle around them or not. 1 is yes, 0 is no."),
+					{ TYPE_INT }, (TYPE_INT),
+					&Thread::GetCardLocked
+				},
+				{
+					94, EXPRCAT_CHARPROP,
+					TEXT("FapState"), TEXT("%p ::FapState"), TEXT("Returns whether the character is fapping or not. 1 is yes, -1 is no."),
```
Comment (geneishouko) [inline AAUnlimited/Functions/Shared/Triggers/Expressions.cpp]: (!) SetMasturbating
Get and Set use different terms

File: AAUnlimited/Functions/Shared/Triggers/Actions.cpp [hunk 6]
```diff
@@ -1521,6 +1552,24 @@ namespace Shared {
...
+				&Thread::IsConsensualH
+			},
+			{
+				77, ACTIONCAT_NPCACTION, TEXT("Lock character"), TEXT("%p ::LockState = %p"),
+				TEXT("Locks the character from interacting. Adds the red circle around them. 1 - set red circle, 0 - unset it."),
+				{ TYPE_INT, TYPE_INT },
+					&Thread::SetCharacterLocked
+			},
+			{
+				78, ACTIONCAT_NPCACTION, TEXT("Set fap state"), TEXT("%p ::FapState = %p"),
```
Comment (geneishouko) [inline AAUnlimited/Functions/Shared/Triggers/Actions.cpp]: (!) SetMasturbating
Get and Set use different terms

File: AAUnlimited/Functions/Shared/Triggers/Actions.cpp [hunk 7]
```diff
@@ -1521,6 +1552,24 @@ namespace Shared {
 				{ TYPE_INT, TYPE_INT, TYPE_BOOL },
 				&Thread::SetLover
 			},
+			{
+				76, ACTIONCAT_NPCACTION, TEXT("Sex Sonsensual"), TEXT("SexConsensual = %p"),
```
Comment (geneishouko) [inline AAUnlimited/Functions/Shared/Triggers/Actions.cpp]: Typo?

File: AAUnlimited/Functions/Shared/Triggers/Actions.cpp [hunk 8]
```diff
@@ -919,6 +942,14 @@ namespace Shared {
 		}
 
 		//int seatPC, int seatPartner
+
+		void Thread::IsConsensualH(std::vector<Value>& params) {
+			int forceVal = params[0].bVal ? 0 : 4;
+			const DWORD offset[]{ 0x376164, 0x44, 0x14, 0x2C, 0x14, 0x9C };
```
Comment (geneishouko) [inline AAUnlimited/Functions/Shared/Triggers/Actions.cpp]: Where did this come from?

Comment (kayfour) [reply AAUnlimited/Functions/Shared/Triggers/Actions.cpp]: Which part? Not quite sure what you are commenting on

File: AAUnlimited/Functions/Shared/Triggers/Actions.cpp (hunk 7 above)
Comment (kayfour) [reply AAUnlimited/Functions/Shared/Triggers/Actions.cpp]: yes

**Pull request merged**